content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,171 @@
+# CI for ML
+
+## Motivation
+
+ML engineers and data scientists face many unique challenges. Some of these
+challenges can be solved by adopting first-citizen concepts from traditional
+software engineering and development. Probably the most common adoption is
+versioning control. Some other, unfortunately less common, could be unit testing
+data processing steps of a pipeline. Here is a non exhaustive list of a few
+other unique challenges:
+
+- Automatically re-train models upon changes.
+- Compare (and report) performance of different versions of a model.
+- Test data correctness when new data is introduced.
+- Handle a model training that is computationally intensive.
```

Comment:
> Handle a model training...

What is the challenge? Not sure I understand this one 🙂

Reply:
This point is derived from a discussion I had with @shcheklein. It points at the case of a model training that either is very lengthy or depends on some specific environment (e.g. GPU) etc. In this case, the CI flow can run the training on the most suitable environment and free the developer's resources in the meantime. Does it make more sense?

Reply:
I take it that this one can be resolved.

Reply:
@jorgeorpinel Can resolve this one?

Reply:
Yes. Sorry took so long, got distracted with a million things. Feel free to resolve the stuff you've addressed. Resolving this one. Thanks